crates/biome_graphql_parser/src/parser/definitions/schema.rs

```diff
@@ -0,0 +1,123 @@
+use crate::parser::{
+    directive::DirectiveList,
+    parse_description,
+    parse_error::{
+        expected_named_type, expected_operation_type, expected_root_operation_type_definition,
+    },
+    r#type::parse_named_type,
+    value::is_at_string,
+    GraphqlParser,
+};
+use biome_graphql_syntax::{
+    GraphqlSyntaxKind::{self, *},
+    T,
+};
+use biome_parser::{
+    parse_lists::ParseNodeList, parse_recovery::ParseRecovery, parsed_syntax::ParsedSyntax,
+    prelude::ParsedSyntax::*, Parser,
+};
+
+use super::{is_at_definition, operation::OPERATION_TYPE};
+
+#[inline]
+pub(crate) fn parse_schema_definition(p: &mut GraphqlParser) -> ParsedSyntax {
+    if !is_at_schema_definition(p) {
+        return Absent;
+    }
+    let m = p.start();
+    // description is optional
+    parse_description(p).ok();
+
+    p.bump(T![schema]);
+
+    DirectiveList.parse_list(p);
+    p.expect(T!['{']);
+    RootOperationTypeDefinitionList.parse_list(p);
+    p.expect(T!['}']);
+
+    Present(m.complete(p, GRAPHQL_SCHEMA_DEFINITION))
+}
+
+#[derive(Default)]
+struct RootOperationTypeDefinitionList;
+
+impl ParseNodeList for RootOperationTypeDefinitionList {
+    type Kind = GraphqlSyntaxKind;
+    type Parser<'source> = GraphqlParser<'source>;
+
+    const LIST_KIND: Self::Kind = GRAPHQL_ROOT_OPERATION_TYPE_DEFINITION_LIST;
+
+    fn parse_element(&mut self, p: &mut Self::Parser<'_>) -> ParsedSyntax {
+        parse_root_operation_type_definition(p)
+    }
+
+    fn is_at_list_end(&self, p: &mut Self::Parser<'_>) -> bool {
+        is_at_root_operation_type_definition_end(p)
```

Comment:
Could you please help me to understand why we need here `is_at_root_operation_type_definition_end`?
What cases do we cover? Is it possible to check only `p.at(T!['}'])`?

Reply:
Do you think this is a good idea?

Reply:
I believe it's a great idea.

I'm wondering if it's possible to have a 'complex' check when we get an `Absent` result from the `parse_element` function.
Specifically, it would be beneficial to maintain `p.at(T!['}'])` the happy path, avoiding additional checks which are required for the failure path.

I don't have any work solution for now, we might think about it in the next MR.